src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/MonoDroidMarkStep.cs

```diff
@@ -142,6 +143,12 @@ protected override TypeDefinition MarkType (TypeReference reference)
 			if (type.Module.Assembly.Name.Name == "System.Core")
 				ProcessSystemCore (type);
 
+			if (type.HasMethods && type.HasInterfaces && type.Implements (ICustomMarshalerName)) {
+				foreach (MethodDefinition method in type.Methods)
+					if (method.Name == "GetInstance" && method.IsStatic && method.HasParameters && method.Parameters.Count == 1 &&  method.ReturnType.FullName == ICustomMarshalerName && method.Parameters.First ().ParameterType.FullName == "System.String")
```

Comment:
Do we need to check the remaining methods once this `if` has executed? Perhaps we can `break` after the `MarkMethod(method)`?